core/lib/spree/core/engine.rb

```diff
@@ -105,6 +105,12 @@ class Engine < ::Rails::Engine
         ]
       end
 
+      initializer 'spree.promo.register.promotions.shipping_actions', before: :load_config_initializers do |app|
```

Comment:
@kennyadsl here's the case that I'm thinking of:

- Developer creates a `OurGreatShippingAction` promo action and adds it to the list
- Admin creates a promo using that action
- The store decides not to do any _more_ of those promo actions and a developer removes the `OurGreatShippingAction` from the list so it doesn't show up in the admin for _new_ promos anymore.
- However, if this *also* removes the action from `Spree::PromotionHandler::Shipping#active_shipping_promotions` then that means that the existing promo will cease to work as well.

So we need 2 lists -- one list that indicates what actions should be available in the admin, and one list that indicates what promotions to consider in `Spree::PromotionHandler::Shipping#active_shipping_promotions`.

The difference from other promo actions is that other promo actions don't need a special invocation to work correctly -- i.e. other actions don't need a `before_transition from: :delivery, do: :apply_shipping_promotions` in order to work correctly.